src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -73,357 +55,214 @@ public AnnotationConfiguration LoadConfig()
             {
-                // get/set parameters
-                string documentGuid = loadDocumentPageRequest.guid;
-                int pageNumber = loadDocumentPageRequest.page;
-                password = loadDocumentPageRequest.password;
-                PageDescriptionEntity loadedPage = new PageDescriptionEntity();
-                ImageOptions imageOptions = new ImageOptions()
-                {
-                    PageNumber = pageNumber,
-                    CountPagesToConvert = 1,
-                    Password = password
-                };
-                string documentPath = GetDocumentPath(documentGuid);               
-                // get page image
-                byte[] bytes;
-                using (MemoryStream memoryStream = new MemoryStream())
+                using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
                 {
-                    using (Stream document = File.Open(documentGuid, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                    IDocumentInfo info = annotator.Document.GetDocumentInfo();
+
+                    AnnotationBase[] annotations = GetAnnotations(loadDocumentRequest.guid, info.FileType.ToString(), password);
+
+                    description.guid = loadDocumentRequest.guid;
+                    description.supportedAnnotations = new SupportedAnnotations().GetSupportedAnnotations(info.FileType.ToString());
+
+                    List<string> pagesContent = new List<string>();
+
+                    // TODO: execute only if !loadAllPages
+                    pagesContent = GetAllPagesContent(password, documentGuid, info);
+
+                    for (int i = 0; i < info.PageCount; i++)
                     {
-                        List<PageImage> images = AnnotationImageHandler.GetPages(document, imageOptions);
-                        Stream imageStream = images[0].Stream;
-                        
-                        imageStream.Position = 0;
-                        imageStream.CopyTo(memoryStream);
-                        bytes = memoryStream.ToArray();
-                        foreach (PageImage page in images)
+                        PageDataDescriptionEntity page = new PageDataDescriptionEntity
+                        {
+                            number = i + 1,
+                            height = info.PagesInfo[i].Height,
+                            width = info.PagesInfo[i].Width,
+                        };
+
+                        if (annotations != null && annotations.Length > 0)
                         {
-                            page.Stream.Close();
+                            page.SetAnnotations(AnnotationMapper.instance.MapForPage(annotations, i+1, info.PagesInfo[i]));
                         }
+
+                        //PageDataDescriptionEntity pageData = GetPageDescriptionEntities(i + 1);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/groupdocs-annotation/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372471)